x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -585,6 +594,7 @@ public void messageReceived(DataNodeRequest request, TransportChannel channel, T
                 configuration,
                 request.aliasFilters(),
                 ActionListener.wrap(searchContexts -> {
+                    assert ThreadPool.assertCurrentThreadPool(ESQL_THREAD_POOL_NAME);
```

Comment:
Without the fix, the test `EsqlActionIT#testRefreshSearchIdleShards` trips this assertion .

Reply:
:+1: